gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/plan/domain_service/CreatePlanDomainService.java

```diff
@@ -61,30 +64,62 @@ public CreatePlanDomainService(
         this.auditService = auditDomainService;
     }
 
-    public PlanWithFlows create(Plan plan, List<Flow> flows, Api api, AuditInfo auditInfo) {
+    public PlanWithFlows create(Plan plan, Supplier<? extends List<? extends AbstractFlow>> flowsSupplier, Api api, AuditInfo auditInfo) {
         return switch (api.getDefinitionVersion()) {
-            case V4 -> createV4ApiPlan(plan, flows, api, auditInfo);
+            case V4 -> createV4ApiPlan(plan, flowsSupplier.get(), api, auditInfo);
             case FEDERATED -> createFederatedApiPlan(plan, auditInfo);
             default -> throw new IllegalStateException(api.getDefinitionVersion() + " is not supported");
         };
     }
 
-    private PlanWithFlows createV4ApiPlan(Plan plan, List<Flow> flows, Api api, AuditInfo auditInfo) {
+    private PlanWithFlows createV4ApiPlan(Plan plan, List<? extends AbstractFlow> flows, Api api, AuditInfo auditInfo) {
         if (api.isDeprecated()) {
             throw new ApiDeprecatedException(plan.getApiId());
         }
 
-        if (api.getApiDefinitionHttpV4().getListeners() != null) {
+        var listeners = api.getApiListeners();
+
+        if (!listeners.isEmpty()) {
             planValidatorDomainService.validatePlanSecurityAgainstEntrypoints(
                 plan.getPlanSecurity(),
-                api.getApiDefinitionHttpV4().getListeners().stream().map(Listener::getType).toList()
+                listeners.stream().map(AbstractListener::getType).toList()
             );
         }
 
         planValidatorDomainService.validatePlanSecurity(plan, auditInfo.organizationId(), auditInfo.environmentId());
-        planValidatorDomainService.validatePlanTagsAgainstApiTags(plan.getPlanDefinitionHttpV4().getTags(), api.getTags());
+        planValidatorDomainService.validatePlanTagsAgainstApiTags(plan.getTags(), api.getTags());
         planValidatorDomainService.validateGeneralConditionsPageStatus(plan);
 
+        if (api.isNative()) {
+            return createNativeV4ApiPlan(plan, (List<NativeFlow>) flows, api, auditInfo);
+        }
+
+        return createHttpV4ApiPlan(plan, (List<Flow>) flows, api, auditInfo);
+    }
+
+    private PlanWithFlows createNativeV4ApiPlan(Plan plan, List<NativeFlow> flows, Api api, AuditInfo auditInfo) {
+        planValidatorDomainService.validateNativePlanFlows(flows);
```

Comment:
What about use existing validation for this rule : `flowValidationDomainService.validateAndSanitizeNativeV4(flows)`